src/lazy/encoder/annotate.rs

```diff
@@ -1,30 +1,30 @@
-use crate::lazy::encoder::value_writer::ValueWriter;
+use crate::lazy::encoder::annotation_seq::AnnotationSeq;
+use crate::lazy::encoder::value_writer::{AnnotatableWriter, ValueWriter};
 use crate::lazy::encoder::write_as_ion::WriteAsIon;
-use crate::raw_symbol_token_ref::AsRawSymbolTokenRef;
 use crate::IonResult;
 
 /// Associates a value to serialize with a sequence of annotations.
-pub struct Annotated<'a, T: ?Sized, A> {
+pub struct Annotated<'a, T: ?Sized, A: 'a> {
     value: &'a T,
-    annotations: &'a [A],
+    annotations: A,
 }
 
-/// Provides implementors with an extension method ([`annotate`](Annotate::annotated_with)) that allows
+/// Provides implementors with an extension method ([`annotate`](Annotatable::annotated_with)) that allows
 /// them to be serialized with an associated sequence of annotations.    
-pub trait Annotate {
+pub trait Annotatable {
     /// Pairs a reference to the provided value with a slice containing annotations.
     ///
     /// ```
     ///# use ion_rs::IonResult;
     ///# fn main() -> IonResult<()> {
     /// use ion_rs::{Element, IonData};
     /// use ion_rs::lazy::encoder::text::LazyRawTextWriter_1_0;
-    /// use ion_rs::lazy::encoder::annotate::Annotate;
+    /// use ion_rs::lazy::encoder::annotate::Annotatable;
     ///
     /// let mut buffer = vec![];
     /// let mut writer = LazyRawTextWriter_1_0::new(&mut buffer);
     ///
-    /// writer.write(42_usize.annotated_with(&["foo", "bar", "baz"]))?.flush()?;
+    /// writer.write(42_usize.annotated_with(["foo", "bar", "baz"]))?.flush()?;
```

Comment:
Does it have to take ownership of the annotations now? Based on the trait impls for `AnnotationSequence`, it looks like it should still work to borrow the slice here.